hudi-common/src/main/scala/org/apache/hudi/HoodieTableFileIndexBase.scala

```diff
@@ -87,6 +89,12 @@ abstract class HoodieTableFileIndexBase(engineContext: HoodieEngineContext,
 
   refresh0()
 
+  /**
+   * Returns latest completed instant as seen by this instance of the file-index
+   */
+  def latestCompletedInstant(): Option[HoodieInstant] =
```

Comment:
I think this is the approach we do take in Java, but in Scala this seems to be off (since you don't add braces at the end of the file-name so get-prefix feels off.

Reply:
I'm trying to understand if this `def function` is more like a `val` function or method in a class.  If it acts like a function, it can be defined as a `val` function (`val latestCompletedInstant = () => getActiveTimeline.filterCompletedInstants().lastInstant()`) and such naming is fine.  If it's more like a method, `getLatestCompletedInstant()` naming is better.

Reply:
It's def closer to former